### ISSUE TITLE
Updating Node.js from v16.18.0 to v18.15.0

### ISSUE THREAD

**Issue by aafayed** [homebridge/homebridge#3343]
### Current Situation

I have Homebridge installed on a virtual machine using hyperv. I was trying to update the Node.js version from v16.18.0 to the latest version (presumably v18.15.0). I used the command on the installtion guide: sudo hb-service update-node
I'm getting the error:
"Your version of Linux does not meet the GLIBC version requirements to use this tool to upgrade Node.js. Wanted: >=2.28. Installed: 2.27"
My node -v commad yields v16.18.0, and my npm -v commad yields 8.19.2, and my ldd --version command yields ldd (Ubuntu GLIBC 2.27-3ubuntu1.4) 2.27
I saw that v16.x will stop being supported by September 2023, so I'm trying to update ahead of time to avoid my system breaking.
Any hints of how to fix this?

### Logs

```markdown
ℹ Updating Node.js from v16.18.0 to v18.15.0...
✖ Your version of Linux does not meet the GLIBC version requirements to use this tool to upgrade Node.js. Wanted: >=2.28. Installed: 2.27
```

### Configuration

```json
I don't know where to get this file from, but I installed Homebridge using the instructions at:
https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Windows-10-Using-Hyper-V
```

### Environment

- OS: ldd (Ubuntu GLIBC 2.27-3ubuntu1.4) 2.27
- Homebridge: 1.6.0
- Node: v16.18.0
- npm: 8.19.2

### Process Supervisor

hb-service

### Additional Context

I installed Homebridge using the instructions at:
https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Windows-10-Using-Hyper-V

**Comment by Pottebaum** [comment]
I'm having the exact same issue as @aafayed, and while I see this has been auto closed I'm hoping someone has any idea on how to resolve as I have several plugins now warning about needing the newer version of Node.js.  

Environment
OS: Ubuntu Bionic Beaver (18.04.6 LTS)
Homebridge: 1.6.0
Node: [v16.14.0]

root@homebridge-vm:/homebridge# sudo hb-service update-node
ℹ Updating Node.js from v16.14.0 to v18.16.0...
✖ Your version of Linux does not meet the GLIBC version requirements to use this tool to upgrade Node.js. Wanted: >=2.28. Installed: 2.27

Any help is much appreciated!

**Comment by aafayed** [comment]
Hello, I'm not sure why the issue was closed, but I'm still unable to find a solution to this issue. As Pottebaum is indicated, I'm also having a lot of warning and not sure where to go from here. Any help would be appreciated. Shall we open a new ticket, or can this ticket be reopened?

**Comment by Pottebaum** [comment]
Hi @aafayed, I went ahead and created a new VM, per the instructions here: https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Windows-10-Using-Hyper-V and someone must have updated the ISO at some point because the fresh VM was updated to:

```
OS: Ubuntu Bionic Beaver (18.04.6 LTS) -> Ubuntu Focal Fossa (20.04.5 LTS)
Node: v16.14.0-> v18.16.0
```

In addition I was able to do a point update for Node.js using the 'hb-service update-node' command without any issue.

I guess the question is; is there an upgrade path or do we need to create a new VM and then transfer the data to it?  Maybe a backup/restore would do it but I have not tried it yet...

**Comment by Pottebaum** [comment]
I can understand that.  This makes it seem like it will grab all that data: https://github.com/homebridge/homebridge/wiki/Backup-and-Restore.  

`A Homebridge backup generated using the [Homebridge UI] contains the entire contents of your Homebridge storage directory, which includes your Homebridge config, cached accessories, HomeKit pairings, plugin information, and Homebridge UI user accounts.`

I'll probably give this a try over the weekend and I can report back on how it goes.  I have some automations but nothing too extensive.

**Comment by aafayed** [comment]
Hi Pottebaum, yes, I would really, really appreciate it if you let me know how your automations got affected. The problem with Homekit is that if it thinks a device is no longer available, it replaces it in the automation by a blank field. So one can't even tell what the device it was to re-select it again. I even saw that it would delete the automation altogether if it thinks a device in the automation is no longer available. Another issue is that I'm using triggers in Controller in some cases, while having the actions in Homekit. So I will probably need to worry about how Controller will perceive the new VM. However, let's see how Homekit reacts first and take it from there.

**Comment by Pottebaum** [comment]
Ok, the process was not as smooth as I would have liked but I was able to get the new VM created and everything moved over.  Here was my process.

	1. Login to existing HomeBridge instance and ensure it is running 1.6.1 as that is what the new Hyper-V images is running
	2. Download backup file via https://github.com/homebridge/homebridge/wiki/Backup-and-Restore
	3. Note existing HomeBridge IP address, which is fixed in my case
	4. Turn off existing HomeBridge VM
	5. Create new HomeBridge VM via https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Windows-10-Using-Hyper-V
	6. Set new HomeBridge IP to same as the old one in step 3, make it fixed (not sure if this is needed but I have a bunch of firewall rules so easier to keep the same)
	7. After HomeBridge has booted up with the new IP go to website
	8. Select Restore and choose backup file from step 2
		a. This is where things started to fall apart.  
		b. This failed to upload and would just hang on "uploading" known bug per https://github.com/homebridge/homebridge-config-ui-x/issues/1457.  
		c. After a bunch of troubleshooting you have to downgrade the Config-UI plugin.  
		d. The command to do this is per https://github.com/homebridge/homebridge-config-ui-x/issues/1457
		e. npm install -g homebridge-config-ui-x@4.50.1
	9. Once this was done the restore worked exactly how you would expect.  


Good news is everything in both HomeBridge and HomeKit are exactly the same as before.  All my accessories and automations are still in HomeKit and working.  No re-pairing was needed.  I do use Controller, not for automations, but did create a very simple one in Controller before this process and it seems to still work.  

Also note that once during this process I turned off the new VM and turned back on the old one.  Everything in HomeKit still worked so if you do try this and it bombs out you can just turn off the new VM and turn the original one back on.  I made sure never to have them both turned on at the same time.

**Comment by aafayed** [comment]
Perfect. I will attempt this weekend and report back on what happens. Thank you for all the effort figuring this out!

**Comment by TheAustinG** [comment]
> Perfect. I will attempt this weekend and report back on what happens. Thank you for all the effort figuring this out!

@aafayed Did you end up having any success? Sadly this is still an issue and I haven't found an easy workaround. I'm thinking about trying the backup/restore that was mentioned above but I'm afraid I'll screw something up or it just won't work.

Edit: GOOD NEWS! I am naturally an impatient person so I decided to go ahead and try the method that @Pottebaum outlined above and I was met with success! The best part is, I didn't run into any issues at all. The restore process was smooth and I did not have to downgrade the Config-UI plugin.

**Comment by aafayed** [comment]
@TheAustinG , I did the same thing back then using the procedure suggested by @Pottebaum and it also worked perfectly for me. The only thing that is different is that I did not need to worry about making the old VM with a static IP or making the new VM with a static IP. When I created the new VM and restored from the backup file, the new homebridge was recognized as the original one and everything worked fine. The only thing I needed to update is the IP address of the login page to reflect the new IP address of the new VM, which I got from the router. I have so much automation in this thing and I'm grateful for @Pottebaum for saving the day!
